pkgs/applications/audio/flac2all/default.nix

```diff
@@ -0,0 +1,37 @@
+{ python3Packages, lib, flac, lame, opusTools, vorbis-tools, ffmpeg }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "flac2all";
+  version = "5.1";
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "OBjlr7cbSx2WOIfZUNwHy5Hpb2Fmh3vmZdc70JiWsiI=";
+  };
+
+  # Not sure why this is needed, but setup.py expects this to be set
+  prePatch = ''
+    echo ${version} > ./flac2all_pkg/version
+  '';
+
+  propagatedBuildInputs = [
+    # Hard requirements
+    python3Packages.pyzmq
+    flac
+    lame
+    # Optional deps
+    opusTools
+    vorbis-tools
+    ffmpeg
```

Comment:
From those packages the binaries are used or the shared objects? For the first wrappers are preferred.

Reply:
Binaries - do you have an example of what "wrappers are preferred" means?

Reply:
Everything that uses makeWrapper. https://github.com/NixOS/nixpkgs/blob/5a893e22a00dd17d2318fe67fd56591876d36abb/pkgs/applications/audio/bespokesynth/default.nix#L124-L134

Reply:
Oh thanks, that makes sense.  We want to set our own path rather than rely on the user's path along with propagated inputs.